File: ios/AudioPlayer.m
//
//  AudioPlayer.m
//  AudioPlayer
//
//  Created by Takuya Ohsugi on 2021/03/04.
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(AudioPlayer, NSObject)

RCT_EXTERN_METHOD(addEvent:(NSString *)name location:(NSString *)location date:(nonnull NSNumber *)date)

@end

File: ios/AudioPlayer.swift
//
//  AudioPlayer.swift
//  AudioPlayer
//
//  Created by Takuya Ohsugi on 2021/03/04.
//

import AVFoundation

@objc(AudioPlayer)
class AudioPlayer: NSObject {
  var audioPlayer: AVAudioPlayer!

 @objc(addEvent:location:date:)
 func addEvent(_ name: String, location: String, date: NSNumber) -> Void {
   // Date is ready to use!
 }

 @objc
 func constantsToExport() -> [String: Any]! {
   return ["state": "init"]
 }
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
}
